tests/Spec/CheckoutFinland/PaymentSpec.php

```diff
@@ -154,41 +155,40 @@ function it_calculates_a_mac_from_variables()
             'familyName'        => 'Doe',
             'address'           => 'Some street 13 B 2',
             'postcode'          => '33100',
-            'postOffice'        => 'Some city',
-            
+            'postOffice'        => 'Some city'
         ];
 
-        $mac_string = '';
-        
-        foreach ($payment_data as $value) {
-            $mac_string .= "$value+";
-        }
 
-        $mac_string .= $this->demo_merchant_secret;
+        $hashString = join(
+            '+',
+            array_values($payment_data)
+        );
+
+        $expected_mac = strtoupper(hash_hmac('sha256', $hashString, $this->demo_merchant_secret));
 
-        $expected_mac = strtoupper(md5($mac_string));
 
         $this->setUrls('www.someurl.com');
 
         $this->setOrderData(
-            $payment_data['stamp'], 
-            $payment_data['amount'], 
-            $payment_data['reference'], 
-            $payment_data['message'], 
+            $payment_data['stamp'],
+            $payment_data['amount'],
+            $payment_data['reference'],
+            $payment_data['message'],
             new \DateTime('2014-10-05')
```

Comment:
Datetime here? Was string in Payment.php

Reply:
setDeliveryDate() casts it to a string in the format we require it to be